lmdeploy/serve/openai/api_server.py

```diff
@@ -434,10 +451,21 @@ async def chat_completions_v1(request: ChatCompletionRequest,
         stop_words=request.stop,
         skip_special_tokens=request.skip_special_tokens)
 
+    tools = None
+    if request.tools and request.tool_choice != 'none':
```

Comment:
Shall we make error response to the client?

Reply:
I've made it in the `check_request` function.